chainer/computational_graph.py

```diff
@@ -0,0 +1,201 @@
+        """
+
+        assert isinstance(node, variable.Variable) or\
+            isinstance(node, function.Function)
+        self.node = node
+        self.id_ = id(node)
+        self.attribute = {
+            "label": self.node.label,
+            "shape": self._shape()
+        }
+
+    def _shape(self):
+        """Return shape type of node
+
+        """
+
+        if isinstance(self.node, variable.Variable):
+            return "oval"
+        elif isinstance(self.node, function.Split):
+            return "hexagon"
+        else:
+            return "box"
+
+    @property
+    def label(self):
+        """Return label that represents its property
+
+        Returns:
+            string: label that represents its id and attributes
+        """
+
+        attributes = ["%s=\"%s\"" % (k, v) for (k, v)
+                      in self.attribute.items()]
+        return "%s [%s];" % (self.id_, ",".join(attributes))
+
+
+class ComputationalGraph(object):
+    """Class that represents computational graph
+
+    .. note::
+
+      We assume that computational graph is directed and is a DAG.
+    """
+
+    def __init__(self, edges):
+        """Initialize computational graph
+
+        Args:
+            edges: List of edges. Each edge consists of pair of nodes.
+            Nodes are either :class: `Variable` object or
+            :class: `Function` object.
+        """
+        self.edges = edges
+
+    def _to_dot(self):
+        """Returns graph in dot format
+
+        Returns:
+            string: graph in dot format
+        """
+
+        ret = "digraph graphname{"
+        for edge in self.edges:
+            head, tail = edge
+            assert (isinstance(head, variable.Variable)
+                    and isinstance(tail, function.Function)) or \
+                   (isinstance(head, function.Function)
+                    and isinstance(tail, variable.Variable))
+            head_node = DotNode(head)
+            tail_node = DotNode(tail)
+            ret += head_node.label
+            ret += tail_node.label
+            ret += "%s -> %s;" % (head_node.id_, tail_node.id_)
+        ret += "}"
+        return ret
+
+    def dump(self, format='dot'):
+        """Dump graph as a text
+
+        Args
+            format: specifies graph language to output
+
+        Returns
+            string: graph in specified format
+        """
+        if format == 'dot':
+            return self._to_dot()
+        else:
+            NotImplementedError('Currently, only dot format is supported')
+
+    def __len__(self):
+        return len(self.edges)
+
+    def __contains__(self, e):
+        return e in self.edges
+
+
+def build_computational_graph(outputs, remove_split=True):
+    """Walk back from outputs to get graph whose nodes are reachable from outputs
+
+    Args:
+        outputs: List of nodes. Each node is either :class:`Variable` object
+        or :class:`Function` object.
```

Comment:
Write a type information like "outputs (list):". You also have to write what this argument indicates. The type sometimes does not provide such information, including this case. It is partly included in the summary, though you should write the detail here. For example, how about:
```
outputs (list): Nodes from which the graph is constructed backward.
    Each node must be of type either :class:`Variable` or :class:`Function`.
```
(inserts newlines appropriately)